---
layout: about
title: about
permalink: /
description:

profile:
  align: right
  image: logo_fullsize.png
  address:

news: false  # includes a list of news items
selected_papers: False # includes a list of papers marked as "selected={true}"
social: true  # includes social icons at the bottom of the page
---


**Departmenr of Systems Immunology, Weizmann Institute of Science <br>
Department of Electrical Engineering & Computer Sciences, Center for Computational Biology, UC Berkeley (on leave)**

<!-- <img align="right" height="225px" src="assets/img/logo_fullsize.png"> -->


We are an interdisciplinary group, with interests in statistical machine learning, algorithms, T cell biology and tumor immunology.<br>
We take a data- centric perspective on studying how changes in transcription are associated with molecular and cellular phenotypes in immunity and cancer. To achieve that, we are developing and applying computational tools to design and make sense of large molecular data sets. We have special interest in single cell genomics, with the goal of better understanding of the <a href="https://doi.org/10.1038/nbt.3711">factors</a> that contribute to variability between cells (e.g, 
<a href = "https://doi.org/10.1016/j.cell.2021.05.045"> metabolic activity</a>, 
<a href="https://doi.org/10.1016/j.crmeth.2022.100182"> chromatin structure</a>, 
<a href="https://doi.org/10.1038/s41587-022-01272-8">microenvironment</a>) 
and their broader implications (e.g., in 
<a href="https://doi.org/10.1038/s41467-019-14118-w">autoimmunity</a> and <a href="https://doi.org/10.1016/j.cell.2022.04.015">cancer</a>). <p>

Examples [software]({{ site.baseurl }}{% link _pages/software.md %}) projects in the lab include:

- [scvi-tools]({{ site.baseurl }}{% link _software/scvi-tools.markdown %}) for end-to-end analysis of single-cell omics data with deep generative models
- [cassiopeia]({{ site.baseurl }}{% link _software/cassiopeia.markdown %}) for single cell lineage tracing tree reconstruction
- [vision]({{ site.baseurl }}{% link _software/vision.markdown %}) for interactive interpretation of single cell RNA-seq data
- [epitome]({{ site.baseurl }}{% link _software/epitome.markdown %}) for prediction of epigenetic events using chromatin accessibility

 <!--
The goals of the research in the lab are to understand:

- How transcription is regulated in immune cells.
- How changes in transcription affect cell- and tissue-level phenotypes.

Beyond a basic scientific drive to study these fundamental questions, we are motivated by the translational potential of identifying and modulating the causes of transcriptional shifts. Our contribution to this wide area of research is in developing and applying computational tools (algorithms, software) to design, integrate and interpret large molecular data sets that can teach us about the regulation and the downstream effects of the transcriptional process. Some of the [software]({{ site.baseurl }}{% link _pages/software.md %}) we have developed includes:

- [scvi-tools]({{ site.baseurl }}{% link _software/scvi-tools.markdown %}) for end-to-end analysis of single-cell omics data with deep generative models
- [cassiopeia]({{ site.baseurl }}{% link _software/cassiopeia.markdown %}) for single cell lineage tracing 
- [vision]({{ site.baseurl }}{% link _software/vision.markdown %}) for interactive interpretation of single-cell RNA-seq data
- [epitome]({{ site.baseurl }}{% link _software/epitome.markdown %}) for prediction of epigenetic events using chromatin accessibility

The lab currently focuses on two related areas of research:

- [Modeling](https://www.nature.com/articles/s41592-018-0229-2) and [interpreting](https://nature-research-under-consideration.nature.com/users/37265-nature-communications/posts/48921-integrated-single-cell-analysis-of-blood-and-cerebrospinal-fluid-leukocytes-in-multiple-sclerosis) the effects of heterogeneity between immune cells based on [single-cell genomics](https://www.nature.com/articles/nbt.3711), with a focus on mRNA profiling.
- Exploring regulatory factors that govern cell state transitions based on [large-scale chromatin profiling assays](https://www.ncbi.nlm.nih.gov/pubmed/27789799) and [massively parallel reporter assays](https://www.cell.com/cell-stem-cell/pdfExtended/S1934-5909(19)30421-7).

We pursue these areas primarily in the context of immune cells, covering various aspects of their biology, such as their [differentiation](http://www.nature.com/nature/journal/v496/n7446/full/nature11981.html), [stability](http://www.nature.com/nature/journal/vaop/ncurrent/full/nature11984.html)/ plasticity, and response to acute [stimulation](http://www.cell.com/molecular-cell/retrieve/pii/S1097276512006570).
-->

<!-- The lab is part of the Department of Electrical Engineering & Computer Sciences and the Center for Computational Biology at UC Berkeley. -->


<!-- Write your biography here. Tell the world about yourself. Link to your favorite [subreddit](http://reddit.com){:target="\_blank"}. You can put a picture in, too. The code is already in, just name your picture `prof_pic.jpg` and put it in the `img/` folder.

Put your address / P.O. box / other info right below your picture. You can also disable any these elements by editing `profile` property of the YAML header of your `_pages/about.md`. Edit `_bibliography/papers.bib` and Jekyll will render your [publications page](/al-folio/publications/) automatically.

Link to your social media connections, too. This theme is set up to use [Font Awesome icons](http://fortawesome.github.io/Font-Awesome/){:target="\_blank"} and [Academicons](https://jpswalsh.github.io/academicons/){:target="\_blank"}, like the ones below. Add your Facebook, Twitter, LinkedIn, Google Scholar, or just disable all of them. -->
