---
layout: page
title: ImpulseDE
description: Differential expression for time-course data.
img: /assets/project_img/impulsede.png
importance: 2
---


### DE with an impulse model

We have developed and implemented two differential expression algorithms tailored to time course bulk sequencing data: ImpulseDE and ImpulseDE2. The two methods are based on different statistical frameworks and are appropriate for different situations: ImpulseDE is based on a model-free error model and can be used for any type of time course measurement and may be used for micro array data for example. ImpulseDE2 is tailored to sequencing data (expected counts or counts) and therefore appropriate for RNA-seq, ATAC-seq and ChIP-seq.

We have recently developed [LineagePulse](https://github.com/YosefLab/LineagePulse), which extends the model of ImpulseDE2 to identify genes that are differentially expressed in an ordered set of single cells.  LineagePulse is appropriate for use in singe cell RNA-seq data where cells are ordered according to predicted pseudotime.

### ImpulseDE2

ImpulseDE2 can find non-constant expression trajectories which follow a unimodal (impulse) pattern or find trajectories which differ between two conditions (such as case and control). Moreover, ImpulseDE2 can find transiently up or down-regulated genes.

ImpulseDE2 is an R Bioconductor package that can be accessed here. Detailed instructions for usage can be found in the vignette.

Manuscript can be obtained from [here](http://biorxiv.org/content/early/2017/03/05/113548).

### ImpulseDE

ImpulseDE is an R Bioconductor package  for detecting differentially expressed genes whether for a single time course (analysis of differential expression behavior over time) or between two time courses (comparative analysis). It is based on the impulse model introduced by Chechik and Koller in 2009, which captures the impulse-like changes of gene expression typically observed for cells responding to perturbations or environmental changes.

<img src="/assets/project_img/impulsedebanner.png" alt="Banner" width="800"/>


### Objective

To test for differential expression in case of a single time course experiment, we evaluate the extent by which the impulse model (representing the alternative hypothesis) fits the expression profile better than a flat line (null hypothesis). In the case of two time courses (case and control) we evaluate the extent to which two models (a separate model for each time course; representing the alternative hypothesis) describe the expression profile better than a single model (null hypothesis; for example, both data sets are generated by the same model (combined)).

### Input requirements

- any kind or normalized high throughput expression data set, including microarray and RNA-Seq gene expression data as well as ChiP-seq data
- at least six time points (required for model inference)

### Workflow

1. The genes are clustered (separately for each time course) using k-means, where k is determined iteratively to optimize the similarity of gene patterns within a cluster.
2. The parameters of the impulse model are fit to the mean expression profile of each cluster (minimizing the sum of squared error; SSE) using two different optimization strategies. Since both approaches return local optima, the analysis is repeated 50 times (default) for each strategy using different initializations for the parametersâ€™ values.
3. The top three sets of parameters (out of the 100, default) for each cluster are used as starting points to fit the model to each gene separately.
4. Random sampling and bootstrapping is used (Storey et al. , 2005) in order to enable the determination of significance levels (evidences to reject the null hypothesis).
5. The resulting p-values are FDR-corrected (q-value) to account for multiple testing (Benjamini and Hochberg, 1995).

### Features

- The implementation includes the use of multi-threading (running fits to different clusters or different randomized samples in parallel) to further reduce running time

### Availability

Manuscript can be obtained [here](https://oup.silverchair-cdn.com/oup/backfile/Content_public/Journal/bioinformatics/33/5/10.1093_bioinformatics_btw665/2/btw665.pdf?Expires=1489363127&Signature=FZ7daxPpG-Jus-6WQdJveKpjnjfuT7oiCB4Lt42eMMh5G1Yn~puWwFVcu34oBtYj9-Aof4OkHqaRBGo5Sa6-YrgMdImI3RlLXlYm3k-OBqAPYZLSXu8jF4IpKelBA0ukyg5hEd-VKqcFuuhhLUcjFX2sYNgbGwrDa8OJZNdJoCDHHZAcCYEFTe17-~Q3~O3AdOvbOSSMIUn1axT4HihpwyENZ1K85wUS1NTDmeQH07U879R6vFM~iryAAwqWPL54eCiHYTPRPmCMLtAcNWLbo0RTON9r3enlJcLCuM-7xisSXFlXxmQ8aOHhDyAtikD0b~DJZ0NgevqXl8eAt~mf0Q__&Key-Pair-Id=APKAIUCZBIA4LVPAVW3Q).

Download it on [GitHub](https://github.com/YosefLab/ImpulseDE). ImpulseDE is also available on Bioconductor.
<br>
<br>

<h3 class="year">Relevant publications</h3><hr>

{% for paper in site.data.publications %}

{% if paper.impulse == true %}
<div id = "{{ paper.title | replace: ' ', '-' | remove: '.' }}" class="clearfix" width="100%" style="padding-top: 5px; padding-bottom: 25px; clear: both;">
<div valign="top" style="overflow: hidden">
  <b>{{paper.title}}</b><br>
  {{paper.authors | markdownify | remove: '<p>' | remove: '</p>'}}<br>
  <i>{{paper.details | markdownify | remove: '<p>' | remove: '</p>'}}</i>
  , {{paper.year}}<br>
  {% if paper.openaccess %}<i class="ai ai-open-access ai-fw"></i> <a href="{{paper.openaccess}}" target="_blank">Open Access</a><br>{% endif %}
  {% if paper.file %}<i class="far fa-file-alt fa-fw"></i> <a href="{{ paper.file | prepend: '/assets/publications/' | prepend: site.baseurl | prepend: site.url }}" target="_blank">{{paper.filetype}}</a><br>{% endif %}
  {% if paper.pmcid %}<i class="fas fa-landmark fa-fw"></i> <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/{{paper.pmcid}}" target="_blank">PubMed Central {{paper.pmcid}}</a><br>{% endif %}
  {% if paper.contentshare %}<i class="fas fa-door-open fa-fw"></i> <a href="{{paper.contentshare}}" target="_blank">Content Sharing link</a><br>{% endif %}
  {% if paper.doi %}<i class="ai ai-doi ai-fw"></i> <a href="https://doi.org/{{paper.doi}}" target="_blank">{{paper.doi}}</a><br>{% endif %}
  {% if paper.github %}<i class="fab fa-github fa-fw"></i> <a href="{{paper.github}}" target="_blank">GitHub</a><br>{% endif %}
  {% if paper.biorxiv %}<i class="ai ai-biorxiv ai-fw"></i> <a href="{{paper.biorxiv}}" target="_blank">bioRxiv preprint</a><br>{% endif %}
  {% if paper.arxiv %}<i class="ai ai-arxiv ai-fw"></i> <a href="{{paper.arxiv}}" target="_blank">arXiv preprint</a><br>{% endif %}
  {% if paper.preregistered %}<i class="ai ai-preregistered ai-fw"></i> <a href="{{paper.preregistered}}" target="_blank">Preregistration</a><br>{% endif %}
  {% if paper.opendata %}<i class="ai ai-open-data ai-fw"></i> <a href="{{paper.opendata}}" target="_blank">Open Data</a><br>{% endif %}
  {% if paper.openmaterials %}<i class="ai ai-open-materials ai-fw"></i> <a href="{{paper.openmaterials}}" target="_blank">Reproducibility code</a><br>{% endif %}
  {% if paper.newsandviews %}<i class="ai ai-conversation ai-fw"></i> <a href="{{paper.newsandviews}}" target="_blank">News and Views</a><br>{% endif %}
    </div>
</div>

{% endif %}
{% endfor %}